core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.ts

```diff
@@ -18,105 +18,121 @@
+
+              // Delete from the ReadOnlyExplorationBackendApiService's cache
+              // As the two versions of the data (learner and editor) now differ
+              ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+                explorationId, exploration);
+
+              if (successCallback) {
+                successCallback(exploration);
+              }
+            }, function(errorResponse) {
+              if (errorCallback) {
+                errorCallback(errorResponse.data);
+              }
+            }
+          );
         }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
+      );
     };
 
     var _deleteExploration = function(
         explorationId, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(explorationId, null);
-
-      $http['delete'](editableExplorationDataUrl).then(function() {
-        // Delete item from the ReadOnlyExplorationBackendApiService's cache
-        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
-          explorationId);
-        if (successCallback) {
-          successCallback({});
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
+      _getExplorationUrlAsync(explorationId, null).then(
+        function(editableExplorationDataUrl) {
+          $http['delete'](editableExplorationDataUrl).then(function() {
+            // Delete item from the ReadOnlyExplorationBackendApiService's cache
+            ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+              explorationId);
+            if (successCallback) {
+              successCallback({});
+            }
+          }, function(errorResponse) {
+            if (errorCallback) {
+              errorCallback(errorResponse.data);
+            }
+          });
         }
-      });
+      );
     };
 
-    var _getExplorationUrl = function(explorationId, applyDraft) {
+    var _getExplorationUrlAsync = function(explorationId, applyDraft) {
       if (applyDraft) {
-        return UrlInterpolationService.interpolateUrl(
+        return $q.resolve(UrlInterpolationService.interpolateUrl(
           EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE, {
             exploration_id: explorationId,
             apply_draft: JSON.stringify(applyDraft)
-          });
-      }
-      if (!GLOBALS.can_edit && GLOBALS.can_voiceover) {
-        return UrlInterpolationService.interpolateUrl(
-          VOICEOVER_EXPLORATION_DATA_URL_TEMPLATE, {
-            exploration_id: explorationId
-          });
+          })
+        );
       }
-
-      return UrlInterpolationService.interpolateUrl(
-        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
-          exploration_id: explorationId
-        });
+      return UserExplorationPermissionsService.getPermissionsAsync().then(
```

Comment:
Can we not have _getExplorationUrlAsync() be async, e.g. by having a single URL to receive the call? I thought that was the whole point of consolidating some of the backend handlers, and this is what I was referring to in my comment [here](https://github.com/oppia/oppia/pull/7194/files#r316007810).

Reply:
Okay,
Done.